src/CMakeLists.txt

```diff
@@ -424,6 +424,7 @@ target_compile_options(
         Halide
         PRIVATE
         $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Wall>
+        $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Werror>
```

Comment:
I'm a little surprised we didn't already have this, if I'm understanding "PRIVATE" correctly. @alexreinking does this make sense?

Reply:
It does not make sense to do this. One should _never_ hard-code `-Werror` in a `CMakeLists.txt` file. We just don't have enough control over downstream toolchains as an open source project for this to be safe.

This should be set in CI/buildbots via the [`CMAKE_COMPILE_WARNING_AS_ERROR`](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILE_WARNING_AS_ERROR.html) cache variable.

Reply:
This change was meant to be local-only, to flush out the instances.

(Alex and I have a longstanding disagreement over the value of `-Werror`, but this is not the place to resolve it :-)